packages/web/components/screen-manager.tsx

```diff
@@ -0,0 +1,169 @@
+import { runIfFn } from "~/utils/function";
+import { createContext } from "~/utils/react-context";
+
+export interface ScreenManagerState {
+  currentScreen: string;
+  setCurrentScreen: (screen: string) => void;
+  goBack: () => void;
+  canGoBack: boolean;
+}
+
+export const [ScreenManagerProvider, useScreenManager] =
+  createContext<ScreenManagerState>({
+    strict: true,
+    name: "ScreenManager",
+  });
+
+interface ScreenManagerProps {
+  defaultScreen?: string;
+  currentScreen?: string;
+  children: React.ReactNode | ((arg: ScreenManagerState) => React.ReactNode);
+}
+
+/**
+ * `ScreenManager` provides a context for managing screens within a React application,
+ * facilitating screen transitions, backward navigation, and the ability to check
+ * if backward navigation is possible. It leverages a stack to maintain the history of screens,
+ * making it particularly useful for workflows such as multi-step forms or navigation-based interfaces.
+ *
+ * Props:
+ * - `defaultScreen` (optional): Specifies the initial screen to be displayed.
+ * - `currentScreen` (optional): Controls the current screen. When provided, this prop turns the component into a controlled component.
+ * - `children`: Either the content to be rendered or a function that returns content, based on the `ScreenManagerState`.
+ *
+ * @example usage:
+ *
+ * ```jsx
+ * const App = () => (
+ *   <ScreenManager defaultScreen="home">
+ *     <Screen screenName="home">Home Screen</Screen>
+ *     <Screen screenName="about">About Screen</Screen>
+ *   </ScreenManager>
+ * );
+ * ```
+ *
+ * In this example, `ScreenManager` is utilized to manage two screens, "home" and "about",
+ * displaying only the content of the active screen.
+ */
+export const ScreenManager = ({
+  defaultScreen,
+  currentScreen: currentScreenProp,
+  ...otherProps
+}: ScreenManagerProps) => {
+  const [history, { push, pop, peek }] = useStack<string>();
+  const [currentScreen, setCurrentScreen] = useControllableState({
+    defaultValue: defaultScreen,
+    value: currentScreenProp,
+    onChange: () => {
+      push(currentScreen);
+    },
+  });
+
+  const goBack = useCallback(() => {
+    const previousScreen = peek();
+    if (previousScreen) {
+      setCurrentScreen(previousScreen);
+      pop();
+    }
+  }, [peek, pop, setCurrentScreen]);
+
+  const canGoBack = useMemo(() => history.length > 0, [history.length]);
+
+  const context = useMemo(
+    () => ({ currentScreen, setCurrentScreen, goBack, canGoBack }),
+    [canGoBack, currentScreen, goBack, setCurrentScreen]
+  );
+
+  return (
+    <ScreenManagerProvider value={context} {...otherProps}>
+      {runIfFn(otherProps.children, context)}
+    </ScreenManagerProvider>
+  );
+};
+
+interface ScreenProps {
+  screenName: string;
+  children: React.ReactNode | ((arg: ScreenManagerState) => React.ReactNode);
+}
+
+/**
+ * `ScreenManager` provides a context for managing screens within a React application,
```

Comment:
didn't you include this note already above?

Reply:
Fixed on my latest commit. Thank you for the review!